src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTransformer.java

```diff
@@ -165,6 +176,82 @@ private StaticLangXApiView generateXApi(SurfaceTransformerContext context) {
     return xapiClass.build();
   }
 
+  private List<StaticLangPagedResponseView> generatePagedResponseWrappers(
+      SurfaceTransformerContext context) {
+    List<StaticLangPagedResponseView> pagedResponseWrappers = new ArrayList<>();
+
+    for (Method method : context.getSupportedMethods()) {
+      MethodTransformerContext methodContext = context.asMethodContext(method);
+      MethodConfig methodConfig = context.getMethodConfig(method);
+      ModelTypeTable typeTable = context.getTypeTable();
+
+      if (methodConfig.isPageStreaming()) {
+        StaticLangPagedResponseView.Builder pagedResponseWrapper =
+            StaticLangPagedResponseView.newBuilder();
+
+        pagedResponseWrapper.templateFileName(PAGE_STREAMING_RESPONSE_TEMPLATE_FILENAME);
+
+        Field resourceField = methodConfig.getPageStreaming().getResourcesField();
+
+        String pagedResponseTypeName =
+            context
+                .getNamer()
+                .getAndSavePagedResponseTypeName(
+                    method,
+                    context.getFeatureConfig(),
+                    typeTable,
+                    method.getInputType(),
+                    method.getOutputType(),
+                    resourceField);
+
+        pagedResponseWrapper.packageName(context.getApiConfig().getPackageName());
+        pagedResponseWrapper.name(pagedResponseTypeName);
+        pagedResponseWrapper.requestTypeName(
+            typeTable.getAndSaveNicknameFor(method.getInputType()));
+        pagedResponseWrapper.responseTypeName(
+            typeTable.getAndSaveNicknameFor(method.getOutputType()));
+        pagedResponseWrapper.resourceTypeName(
+            typeTable.getAndSaveNicknameForElementType(resourceField.getType()));
+        pagedResponseWrapper.iterateMethods(getIterateMethods(methodContext));
+        pagedResponseWrapper.imports(new ArrayList<ImportTypeView>());
+
+        String outputPath =
+            pathMapper.getOutputPath(context.getInterface(), context.getApiConfig());
+        pagedResponseWrapper.outputPath(
+            outputPath + File.separator + pagedResponseTypeName + ".java");
```

Comment:
Hmmm, yes I think this might be a good idea. Another option is to put them all into the same file, possibly as inner classes.